File: src/main/java/br/com/jiratorio/domain/JiraError.java
package br.com.jiratorio.domain;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class JiraError {

    private String message;

    @JsonProperty("status-code")
    private Long statusCode;

    private List<String> errorMessages;

}

File: common-library/src/main/kotlin/br/com/jiratorio/jira/JiraError.kt
package br.com.jiratorio.jira

import br.com.jiratorio.extension.buildList
import com.fasterxml.jackson.annotation.JsonProperty

data class JiraError(

    val message: String?,

    @field:JsonProperty("status-code")
    val statusCode: Long?,

    @field:JsonProperty("errorMessages")
    val messages: List<String>? = null

) {

    val allErrors: List<String>
        get() = buildList {
            message?.let { add(it) }
            messages?.let { addAll(it) }
        }

}
